Tools/Scripts/webkitpy/api_analyzer/driver.py

```diff
@@ -0,0 +1,241 @@
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import os
+import pkgutil
+import shlex
+import time
+from fnmatch import fnmatch
+from importlib.machinery import FileFinder
+from pathlib import Path
+
+from webkitpy import api_analyzer
+from webkitpy.api_analyzer.sdkdb import sdkdb
+from webkitpy.api_analyzer import macho
+
+from webkitpy.port.config import apple_additions
+additions = getattr(apple_additions(), 'api_analyzer', None)
+
+# Some symbols, namely ones that are low-level parts of system libraries and
+# runtimes, are implicitly available.
+ALLOWED_SYMBOLS = set((
+    '_OBJC_METACLASS_$_NSObject',
+    '_OBJC_EHTYPE_$_NSException',
+    # Foundation APIs
+    '_OBJC_CLASS_$_NSConstantArray',
+    '_OBJC_CLASS_$_NSConstantDictionary',
+    '_OBJC_CLASS_$_NSConstantDoubleNumber',
+    '_OBJC_CLASS_$_NSConstantIntegerNumber',
+    '___CFConstantStringClassReference',
+    '___NSArray0__',
+    '___NSDictionary0__',
+    # rdar://79462292
+    '___kCFBooleanFalse',
+    '___kCFBooleanTrue',
+    '___NSArray0__struct',
+    '___NSDictionary0__struct',
+    # C++ std
+    '__ZdlPv',
+    '__Znwm',
+))
+
+ALLOWED_SYMBOL_GLOBS = (
+    # C++ std
+    '__ZS*',
+    '__ZT*',
+    '__ZNS*',
+    '__ZNK*',
+    '__ZN9__gnu_cxx*',
+    # We remove SwiftUI from SDK DBs due to rdar://143449950.
+    '_$s*7SwiftUI*',
+    # rdar://79109142
+    '__swift_FORCE_LOAD_$_*',
+)
+
+# TBDs from the active SDK whose symbols are treated as implicitly available.
+# Pattern strings on the right-hand side select individual libraries from the
+# TBD.
+SDK_ALLOWLIST = {
+    'usr/lib/libobjc.tbd': (),
+    'usr/lib/swift/lib*.tbd': (),
+    'usr/lib/libc++*.tbd': (),
+    'usr/lib/libSystem.B.tbd': ('/usr/lib/system/libsystem_*',
+                                '/usr/lib/system/libcompiler_rt*',
+                                '/usr/lib/system/libunwind*'),
+    'usr/lib/libicucore.A.tbd': (),
+}
+
+
+class TSVReporter:
+    def __init__(self, args: argparse.Namespace):
+        self.n_issues = 0
+        self.seen: set[str] = set()
+        self.print_details = args.details
+
+    def missing_selector(self, name: str, ignore=False):
```

Comment:
Let's make these boolean arguments all keyword-only:

```suggestion
    def missing_selector(self, name: str, *, ignore=False):
```